### PR TITLE
Reusable CI actions

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

**Review by jmank88 — APPROVED** [review]

File: .github/workflows/push-master.yml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+          # general inputs
+          app-name: chainlink-feeds-${{ matrix.package }}
+          publish: "true"
+          # grafana inputs
+          metrics-job-name: cicd-build-publish-artifacts-dev-matrix (${{ matrix.package }})
+          gc-basic-auth: ${{ secrets.GRAFANA_CLOUD_BASIC_AUTH }}
+          gc-host: ${{ secrets.GRAFANA_CLOUD_HOST }}
+          # aws inputs
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-role-arn: ${{ secrets.AWS_OIDC_IAM_ROLE_ARN }}
```
Comment (chainchad) [inline .github/workflows/push-master.yml:122]: We need this if pushing docker images to AWS ECR

File: .github/workflows/push-master.yml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      changesets_published: ${{ steps.cd-release.outputs.published }}
+      changesets_publishedPackages: ${{ steps.cd-release.outputs.publishedPackages }}
+    steps:
+      - name: cd-release
+        id: cd-release
+        uses: smartcontractkit/.github/actions/cicd-changesets@8720f77f170718126aab659bdd32edd7432556b3 # cicd-changesets@0.1.1
+        with:
+          # general inputs
+          git-user: app-token-issuer-data-feeds[bot]
+          git-email: app-token-issuer-data-feeds[bot]@users.noreply.github.com
```
Comment (chainchad) [inline .github/workflows/push-master.yml]: Need to change to a foundations bot?

File: .github/workflows/push-master.yml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+          pnpm-use-cache: false
+          # aws inputs
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-role-arn: ${{ secrets.AWS_ROLE_ARN_GATI_CHANGESETS }}
```
Comment (chainchad) [inline .github/workflows/push-master.yml]: Will need to setup a GATI to allow changesets to push git tags/create releases/etc

**Review by chainchad — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request-master.yml [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        # Convert "producer,reports_consumer" to "./producer/... ./reports_consumer/..."
+        run: echo "lint_args_packages=$(echo "./$(echo $PACKAGES | sed 's/,/\/... .\//g;s/$/\/.../')")" | tee -a "${GITHUB_OUTPUT}"
+
+  ci-lint:
+    runs-on: ubuntu-latest
+    needs: [init]
+    permissions:
+      id-token: write
+      contents: read
+      actions: read
```
[the remark this answers is not in the document]
Comment (vyzaldysanchez) [reply .github/workflows/pull-request-master.yml:49]: By entirely you mean the `permissions` section?

[the remark this answers is not in the document]
Comment (chainchad) [reply .github/workflows/pull-request-master.yml:49]: Yeah, the [default perms](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) should work if we're not needing to assume an AWS IAM role.

**Pull request merged**